test/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+if (CMAKE_BUILD_TYPE STREQUAL Test)
+    set(TEST_BUILD_TYPE --test)
+elseif (CMAKE_BUILD_TYPE STREQUAL Debug)
+    set(TEST_BUILD_TYPE --debug)
+endif ()
+
+add_custom_target(check
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/runtests.py ${TEST_BUILD_TYPE} --binary ${CMAKE_BINARY_DIR}/ch --logfile ${CMAKE_BINARY_DIR}/check.log
```

Comment:
If I try and use `make test` to run this test several of the tests segfault - tests which pass if I run `./runtests.py -t` Not sure what the cause of this is.

Reply:
Try `./runtests.py -d` - test script has both modes for some reason, or try building with `-DCMAKE_BUILD_TYPE=Test`

Reply:
Using test build type would be a better emulation of what `./build.sh` does. We can also set that to be default build type (the build script defaults to `Release`).

Reply:
I found an issue with how Intl was enabled in CMake build, I think I have a fix, though there might be a few more problems.

By the way, how are you building ICU? I am making this PR default to 'system' ICU, which does not seem to work on Mac. It would be possible to migrate embedded ICU to CMake as well, but that would be handled in future installments.

Reply:
I use the --embed-icu option with build.sh